### DISCUSSION TITLE
Impact of Adstock Normalization

### DISCUSSION THREAD

**Discussion by PK1706** [pymc-labs/pymc-marketing#718] [Q&A] [answered]
Hi,
I am a bit confused in respect to the normalization of the adstock transformation. 
What is the general idea behind it and what implication does enabling/disabling it have?
Lets say we have the following example using the geometric_adstock with **alpha = 0.4** and **l_max = 3**:

- **original** array = [10, 10, 10, 0, 10]
  - sum: 40
- adstocked array (**without normalization**) = [10, 14, 15.6, 5.6, 11.6]
  - sum: 56.80
  - weight arrays: [1, 0.4, 0.16, 0, 0]
- adstocked array (**with normalization**) = [6.41,  8.97, 10, 3.59, 7.44]
  - sum: 36.41
  - weight arrays: [0.64, 0.27, 0.10, 0, 0]
  
From this example and the discussion in the LightweightMMM Github([What is the purpose of normalization in the adstock function? #79](https://github.com/google/lightweight_mmm/discussions/79%20)) it seems like the purpose of normalization is to reduce the buildup of adstock and to keep the scale of adstocked spends comparable to the original spends?

In what way does the normalization however have an impact on the model fitting as well the results? Would normalized vs. non-normalized weights result in different channel contributions?

Also what impact does normalization have on the interpretability?
On the one hand it seems more intuitive to have adstocked spends which are closer to original spends, when using the adstocked spends in the budget optimization (as planned in [User-defined media transformations and custom ordering #632](https://github.com/pymc-labs/pymc-marketing/pull/632)). On the other hand the intuition behind adstocked values being larger than original values due to the incorporation of the carryover effect gets lost. 

As I haven't found any documentation on the normalization except for a short mention in this [blog post](https://juanitorduz.github.io/pymc_mmm/) by @juanitorduz , I would highly appreciate a heads up on this.
Thanks a lot!

**Comment by williambdean** [comment, marked as answer]
In my mind, the normalized version just spreads the media variable through time. This is because the weight array will sum up to 1 when using normalized = True.  How much of today's spend is actually used today's? 64% Tomorrow? 27%

If the `x` vector is extended a bit, you can see that the sum of the normalized one is exactly the sum of the input `x` . i.e. 
```python
x = np.array([10, 10, 10, 0, 10])
x = np.concatenate([x, np.zeros(10)])

alpha = 0.4
l_max = 3
y_norm = geometric_adstock(x, alpha=alpha, l_max=l_max, normalize=True)
y_unnorm = geometric_adstock(x, alpha=alpha, l_max=l_max, normalize=False)

print(y_norm.sum().eval()) # 40.0
print(y_unnorm.sum().eval()) # 62.4
```

The shapes of the two are similar
![adstock-impact](https://github.com/pymc-labs/pymc-marketing/assets/57733339/92d46661-832b-47d3-aa19-59ae3e002c6d)

But I find that the normalization helps with interpretability!

Note that in the models the adstock would most likely not be the final transformation, so them being inputs into saturation and scaling will affect the parameter estimates for subsequent transformations. Kind of like having different units in a regression model.

Maybe Juan can weigh in on his thoughts while writing the article

**Reply by PK1706** [reply]
Thanks a lot, that already clarifies most of my concerns!

To recap, this means that:
- non-normalized adstock maintains the original spend in the first time period and then deteriorates in the l_max next periods
- normalized adstock distributes the original spend across the all l_max periods

I think it would be beneficial to have this clarified in the mmm_example.ipynb, where the synthetic data is transformed.

I am however still not certain about the effect it will have on the results. I am aware of the saturation transformation that gets applied afterwards. How do you deal with this in practice? Do you recommend to fit a model with normalized adstock and one with non-normalized adstock and then compare the results or do you stick to normalized adstock from the beginning?

**Reply by williambdean** [reply]
Hi @PK1706 

I will add an issue to add more about it in the mmm_example notebook. (#783)

My gut is that any performance with normalize=True will be similar to normalize=False. Thus, I'd likely just fit *one* model and stick with normalize=True because of the interpretation. Maybe you can try it out and report your findings!
